src/interpreter/Interpreter.ts

```diff
@@ -54,24 +54,23 @@ export class Interpreter {
   }
 
   public evaluateAst(ast: Ast, formulaAddress: SimpleCellAddress): InterpreterValue {
-    const val = this.evaluateAstWithoutPostoprocessing(ast, formulaAddress)
+    let val = this.evaluateAstWithoutPostoprocessing(ast, formulaAddress)
     if (typeof val === 'number') {
       if (isNumberOverflow(val)) {
-        return new CellError(ErrorType.NUM, ErrorMessage.NaN)
+        val = new CellError(ErrorType.NUM, ErrorMessage.NaN)
```

Comment:
sure, but the reason behind `wrapperForAddress` is to fix whatever comes from evaluation, even in nested, recursive calls

Reply:
Shouldn't all functions return the correct error? We're patching the error object just to avoid fixing this in the right places or there is some way to create an error that don't know it's origin?

Reply:
I've refactored it slightly to make the field private.
To answer your question:
- I'm not sure whether in every place the constructor of CellError is called, we have formulaAddress in the context (there might be some obscure case that would require ugly refactor)
- Adding it here, in a single place, is a minimal change to code that made this feature work, and thus it's much easier to make sure the logic is correctly added in every evaluation.

Reply:
getter/setter are ok, but my main idea was that `CellError` is immutable. 

Once created it cannot be altered, especially the origin address. Having this editable is IMO a risk.

Reply:
It's not getter/setter, its getter and (setter that only sets it once, if the value is not present yet). So I don't see any risk here. Also, keep in mind that this value is supplementary one, it does not affect outcome of calculation, only provides debugging info.